lib/localparticipant.js

```diff
@@ -344,6 +345,16 @@ class LocalParticipant extends Participant {
       }
 
       trackSignaling.on('updated', updated);
+
+      trackSignaling.on('warning', twilioWarningName => {
+        localTrackPublication.emit('warning', twilioWarningName);
+        this.emit('trackWarning', twilioWarningName, localTrack);
+      });
+
+      trackSignaling.on('warningsCleared', () => {
+        localTrackPublication.emit('warningsCleared');
+        this.emit('trackWarningsCleared', localTrack);
+      });
```

Comment:
I think this should be handled within LocalTrackPublication, since it has access to the signaling object.